src/metabase/api/field.clj

```diff
@@ -396,53 +396,56 @@
 
       [<value-of-field> <value-of-remapped-field>]
 
    if a match is found.
 
    For example, with the Sample Database, you could find the name of the Person with ID 20 as follows:
 
-      (remapped-value <PEOPLE.ID Field> <PEOPLE.NAME Field> 20)
-      ;; -> [20 \"Peter Watsica\"]"
-  [field remapped-field value]
+      (remapped-values <PEOPLE.ID Field> <PEOPLE.NAME Field> [20])
+      ;; -> [[20 \"Peter Watsica\"]]"
+  [field remapped-field values]
   (try
     (let [field   (follow-fks field)
           results (qp/process-query
                    {:database (db-id field)
                     :type     :query
                     :query    {:source-table (table-id field)
-                               :filter       [:= [:field (u/the-id field) nil] value]
+                               :filter       (concat [:= [:field (u/the-id field) nil]] values)
                                :fields       [[:field (u/the-id field) nil]
                                               [:field (u/the-id remapped-field) nil]]
-                               :limit        1}})]
-      ;; return first row if it exists
-      (first (get-in results [:data :rows])))
+                               :limit (count values)}})]
+      (get-in results [:data :rows]))
     ;; as with fn above this error can usually be safely ignored which is why log level is log/debug
     (catch Throwable e
       (log/debug e "Error searching for remapping")
-      nil)))
+      [])))
 
-(defn parse-query-param-value-for-field
-  "Parse a `value` passed as a URL query param in a way appropriate for the `field` it belongs to. E.g. for text Fields
+(defn parse-query-param-values-for-field
+  "Parse `values` passed as a URL query param in a way appropriate for the `field` it belongs to. E.g. for text Fields
   the value doesn't need to be parsed; for numeric Fields we should parse it as a number."
-  [field ^String value]
+  [field values]
   (if (isa? (:base_type field) :type/Number)
-    (.parse (NumberFormat/getInstance) value)
-    value))
+    (mapv #(.parse (NumberFormat/getInstance) %) values)
+    values))
 
 (api/defendpoint GET "/:id/remapping/:remapped-id"
   "Fetch remapped Field values."
-  [id remapped-id value]
+  [id remapped-id value values]
   {id          ms/PositiveInt
    remapped-id ms/PositiveInt
-   value       ms/NonBlankString}
+   value       [:maybe ms/NonBlankString] ;; deprecated
+   values      [:maybe (ms/QueryVectorOf ms/NonBlankString)]}
   (let [field          (api/read-check Field id)
         remapped-field (api/read-check Field remapped-id)
-        value          (parse-query-param-value-for-field field value)]
-    (remapped-value field remapped-field value)))
+        values         (parse-query-param-values-for-field field (or values [value]))
+        remapped       (remapped-values field remapped-field values)]
+    (if (some? value) ;; preserve backward compability; previously only one row was returned
```

Comment:
@bshepherdson do you have better ideas to make this backward compatible and more clean?